### PR TITLE
fix(TDI-42092): An error occurs on tFileOutputExcel after Iterate when /tmp/profiles folder does not exist

### PR DESCRIPTION
**Pull request by dmytroshekera**
* Iterate when /tmp/profiles folder does not exist.
* reproduce in multithread env - each thread try create
* folder. First thread create. Second crash because folder exist

**What is the current behavior?** 
We have concurrency issue here. Launch many threads that try create folder. 
So first thread create folder and another fail. There is double check deep inside code before create directory. 

**What is the new behavior?**
Make synchronization on object, that put to ConcurentHashMap if not exist by key "EXCEL_OUTPUT_LOCK_OBJ"

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet [hunk 1]
```diff
@@ -777,7 +777,11 @@ if(!hasDynamic && isIncludeHeader){
 		xlsxTool_<%=cid%>.setRecalculateFormula(<%=recalculateFormula%>);
 		xlsxTool_<%=cid%>.setXY(<%=firstCellYAbsolute%>,<%=firstCellXStr%>,<%=firstCellYStr%>,<%=keepCellFormating%>);
 		<%if(!useStream){%>
-		xlsxTool_<%=cid%>.prepareXlsxFile(fileName_<%=cid%>);
+		java.util.concurrent.ConcurrentHashMap<java.lang.Object, java.lang.Object> chm_<%=cid%> = (java.util.concurrent.ConcurrentHashMap<java.lang.Object, java.lang.Object>) globalMap.get("concurrentHashMap");
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet]: seems "concurrentHashMap" map only when use(from footer.javajet) :

boolean containsTPartitioner = process.getNodesOfType("tPartitioner").size() > 0 ? true : false;
...
java.util.concurrent.ConcurrentHashMap<Object, Object> concurrentHashMap = new java.util.concurrent.ConcurrentHashMap<Object, Object>();
	    globalMap.put("concurrentHashMap", concurrentHashMap);

File: main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet [hunk 2]
```diff
@@ -777,7 +777,11 @@ if(!hasDynamic && isIncludeHeader){
...
+		java.lang.Object lockObj_<%=cid%> = chm_<%=cid%>.computeIfAbsent("EXCEL_OUTPUT_LOCK_OBJ_<%=cid%>", k -> new Object());
+		synchronized (lockObj_<%=cid%>) {
+			xlsxTool_<%=cid%>.prepareXlsxFile(fileName_<%=cid%>);
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet]: i remember i try this steps simplely, seems not work, you can ignore this comment

File: main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet (hunk 1 above)
Comment (wwang-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet]: forget it, i miss the comment : //

Comment (dmytroshekera) [reply main/plugins/org.talend.designer.components.localprovider/components/tFileOutputExcel/tFileOutputExcel_begin.javajet]: talk with @wwang-talend - we have comment  for all if e.g.   //if(containsTPartitioner)

**Pull request merged**